### ISSUE TITLE
Deserialize JSON into a type

### ISSUE THREAD

**Issue by alerickson** [PowerShell/PSResourceGet#140]
Rather than doing all the heavy lifting of parsing yourself... you should just deserialize into a type:

https://www.newtonsoft.com/json/help/html/SerializingJSON.htm#JsonConvert

```csharp
ResourceFileData deserializedProduct = JsonConvert.DeserializeObject<ResourceFileData>(_requiredResourceJson);
```

Then you can have another file for `ResourceFileData.cs` with the type definition there and Newtonsoft will magically convert the Json into an instance of that type.

Let me know if you need help with this or let me know if you didn't want to go down this route for a specific reason.

_Originally posted by @TylerLeonhardt in https://github.com/PowerShell/PowerShellGet/pull/139_

**Comment by TylerLeonhardt** [comment]
This would likely remove this code (with the exception of psd1 logic):

https://github.com/PowerShell/PowerShellGet/blob/c21f21a19f1f8a021d108c28c72dcf21bfb4509b/src/InstallPSResource.cs#L407-L621

**Comment by TylerLeonhardt** [comment]
It might look something like:

```csharp
// read in or get json string

ResourceFileData deserializedProduct = JsonConvert.DeserializeObject<ResourceFileData>(jsonStr);
InstallBeginning(
    deserializedProduct.PkgName,
    deserializedProduct.Version,
    deserializedProduct.Prerelease,
    deserializedProduct.Repository,
    deserializedProduct.Scope,
    deserializedProduct.AcceptLicense,
    deserializedProduct.Quiet,
    deserializedProduct.Reinstall,
    deserializedProduct.Force,
    deserializedProduct.TrustRepository,
    deserializedProduct.NoClobber);
```

**Comment by TylerLeonhardt** [comment]
This StackOverflow shows how you can use a custom JsonSerializer:

https://stackoverflow.com/questions/54815199/newtonsoft-json-serialize-deserialize-nested-property

This is what you can use to handle packages that just have a version string, vs packages that have a full object.

**Comment by alerickson** [comment]
Resolved in  #139
